include/jemalloc/internal/arena_inlines_b.h

```diff
@@ -328,6 +374,9 @@ arena_dalloc(tsdn_t *tsdn, void *ptr, tcache_t *tcache,
 
 	if (likely(alloc_ctx.slab)) {
 		/* Small allocation. */
+		if (arena_dalloc_small_sanity_check(tsdn, ptr)) {
+			return;
+		}
```

Comment:
It feels this fits better in `tcache_dalloc_small`? It's in fact not doing the arena_dalloc_small path. Maybe move everything over there and rename to tcache_dalloc_small_saftey_check?

Reply:
Renamed but the function kept in the same place to avoid circular dependency.